### PR TITLE
Update io.kubernetes:client-java from 1.0.0 to 16.0.2

### PR DESCRIPTION
**Pull request by eamansour**
For https://github.com/galasa-dev/projectmanagement/issues/1083

- Updated build.gradle to use client-java 16.0.2 and removed dependencies that are no longer used by client-java
- Updated included jars in OSGi bundle
- Fixed compilation errors as a result of client-java's new project structure and updated APIs

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd [hunk 1]
```diff
@@ -20,28 +20,23 @@ Import-Package: dev.galasa.framework,\
...
+    commons-compress-1.21.jar; lib:=true,\
+    commons-lang3-3.12.0.jar; lib:=true,\
+    commons-collections4-4.4.jar; lib:=true,\
+    jose4j-0.7.12.jar; lib:=true,\
+    slf4j-api-1.7.36.jar; lib:=true,\
+    protobuf-java-3.21.2.jar; lib:=true,\
+    jsr305-3.0.2.jar; lib:=true,\
+    swagger-annotations-1.6.6.jar; lib:=true,\
+    logging-interceptor-4.9.2.jar; lib:=true,\
+    okhttp-4.9.2.jar; lib:=true,\
```
Comment (hobbit1983) [inline galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd:38]: Just be careful - I've just bumped okhttp in Selenium manager to 4.10 and that had deps on okio and kotlin runtime

**Review by hobbit1983 — CHANGES_REQUESTED** [review]

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd (hunk 1 above)
Comment (eamansour) [reply galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd:41]: Thanks! This has been resolved in https://github.com/galasa-dev/framework/pull/379/commits/7536b9836cba3c6be85e1a1c0498a2f48d339125.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by hobbit1983 — APPROVED** [review]

**Pull request merged**